templates/pkg/resource/sdk.go.tpl

```diff
@@ -273,3 +273,55 @@ func (rm *resourceManager) terminalAWSError(err error) bool {
 	return false
 {{- end }}
 }
+
+{{- if .CRD.GetImmutableFields }}
+// validateImmutableFields validates the immutable fields and set appropriate condition
+func (rm *resourceManager) validateImmutableFields(ko *svcapitypes.{{ .CRD.Names.Camel }},
+	delta *ackcompare.Delta,) {
+	var fields []string;
+
+	for _, diff := range delta.Differences {
+		if diff.Path.Contains("Spec") {
+		{{- range $immutableField := .CRD.GetImmutableFields }}
+			if diff.Path.Contains("{{$immutableField}}") {
+				fields = append(fields,"{{$immutableField}}")
+			}
+		{{- end }}
+		}
+	}
```

Comment:
Couple things about the above....

a) Recommend putting it into a separate helper function called `hasImmutableFieldChanges`
b) The [Update code paths will never fire if the Spec hasn't been modified](https://github.com/aws-controllers-k8s/runtime/blob/0ff9ffa7634a156ab23b4102e200df50e9ece56b/pkg/runtime/reconciler.go#L248), so I think you should be able to simplify the above to just this:

```
{{- range $immutableField := .CRD.GetImmutableFields }}
if delta.DifferentAt("{{$immutableField}}") {
        fields = append(fields,"{{$immutableField}}")
}
{{- end }}
```